python/src/ops.cpp

```diff
@@ -3587,6 +3587,36 @@ void init_ops(nb::module_& m) {
             array: The output containing elements selected from
             ``x`` and ``y``.
       )pbdoc");
+  m.def(
+      "nan_to_num",
+      [](const ScalarOrArray& a,
+         const ScalarOrArray& nan,
+         const ScalarOrArray& posinf,
+         const ScalarOrArray& neginf,
+         StreamOrDevice s) {
+        return nan_to_num(
+            to_array(a), to_array(nan), to_array(posinf), to_array(neginf), s);
+      },
+      nb::arg(),
+      "nan"_a = 0,
+      "posinf"_a = std::numeric_limits<float>::max(),
+      "neginf"_a = -std::numeric_limits<float>::max(),
```

Comment:
I think these should default to `None` and be chosen based on the dtype of `a`. We will presumably need to use different values for `fp16` and `bf16`.